clients/nodejs/cac-client/client.ts

```diff
@@ -0,0 +1,162 @@
+        'cac_get_default_config': [string, ["pointer", "pointer"]],
+
+    });
+
+    constructor(tenantName: string, pollingFrequency: number, cacHostName: string) {
+        if (tenantName == "" || cacHostName == "") {
+            throw Error("tenantName cannot be null or empty")
+        }
+        this.tenant = tenantName;
+        this.pollingFrequency = pollingFrequency;
+        this.cacHostName = cacHostName;
+    }
+
+    public getCACLastErrorMessage(): string | null {
+        return this.rustLib.cac_last_error_message();
+    }
+
+    public getCACLastErrorLength(): number {
+        return this.rustLib.cac_last_error_length()
+    }
+
+    public getCACClient(): ref.Pointer<unknown> {
+        return this.rustLib.cac_get_client(this.tenant);
+    }
+
+    public createNewCACClient(): number {
+        let resp = this.rustLib.cac_new_client(this.tenant, this.pollingFrequency, this.cacHostName);
+        if (resp == 1) {
+            let errorMessage = this.getCACLastErrorMessage();
+            console.error("Some Error Occur while creating new client ", errorMessage);
+        }
+        return resp;
+    }
+
+    public async startCACPollingUpdate() {
+        if (isMainThread && worker) {
+            worker.postMessage(
+                { tenant: this.tenant
+                , event : "startCACPollingUpdate"
+                , pollingFrequency: this.pollingFrequency
+                , cacHostName: this.cacHostName
+                }
+            );
+            return;
+        }
+        if (!isMainThread) {
+            this.rustLib.cac_start_polling_update(this.tenant)
+        }
+    }
+
+    public getCACConfig(filterQuery: Object | null, filterPrefix: string[] | null): string | null {
+        let strFilterQuery = filterQuery == null ? ref.NULL : ref.allocCString(JSON.stringify(filterQuery));
+        let strFilterPrefix = filterPrefix == null ? ref.NULL : ref.allocCString(filterPrefix.join(this.delimeter));
+        let clientPtr = this.getCACClient();
+        return this.rustLib.cac_get_config(clientPtr, strFilterQuery, strFilterPrefix);
+    }
+
+    public freeCACClient(clientPtr: ref.Pointer<unknown>) {
+        this.rustLib.cac_free_client(clientPtr)
+    }
+
+    public freeCACString(str: string) {
+        this.rustLib.cac_free_string(str)
+    }
+
+    public getLastModified(): string | null {
+        return this.rustLib.cac_get_last_modified(this.getCACClient());
+    }
+
+    public getResolvedConfig(query: Object, filterKeys: string[] | null, mergeStrategy: string): string | null {
+        let strQuery = JSON.stringify(query);
+        let strFilterKeys = filterKeys == null ? ref.NULL : ref.allocCString(filterKeys.join(this.delimeter));
+        return this.rustLib.cac_get_resolved_config(
+            this.getCACClient(), strQuery, strFilterKeys, mergeStrategy
+        )
+    }
+    public getDefaultConfig(filterKeys: string[] | null): string | null {
```

Comment:
if the output of the functions are null, then you should check for the error and then return the error caused during the function call (for all functions returning `null` | `some_value`)

Reply:
done